components/brave_welcome_ui/components/select-browser/index.tsx

```diff
@@ -78,10 +78,12 @@ function BrowserItemButton (props: BrowserItemButtonProps) {
 function SelectBrowser () {
   const { browserProfiles, currentSelectedBrowser, setCurrentSelectedBrowser, setViewType, incrementCount, scenes } = React.useContext(DataContext)
   const browserTypes = getUniqueBrowserTypes(browserProfiles ?? [])
-  WelcomeBrowserProxyImpl.getInstance().getDefaultBrowser().then(
-    (name: string) => {
-    setCurrentSelectedBrowser?.(name)
-  })
+  React.useEffect(() => {
```

Comment:
We usually follow a structure inside a component

1. local state
2. local functions
3. side-effects
4. return

Please move useEffect call right before the return. Also add a new line between.

```js
React.useEffect(() => {}, [])

return ()
```